fineract-provider/src/main/java/org/apache/fineract/portfolio/savings/data/DepositAccountDataValidator.java

```diff
@@ -606,9 +606,9 @@ private void validateDepositTermDeatilForUpdate(final JsonElement element, final
 
     private void validateForInterestCalc(final JsonElement element, final DataValidatorBuilder baseDataValidator) {
 
-        Long principalAmount = this.fromApiJsonHelper.extractLongNamed(principalAmountParamName, element);
-        baseDataValidator.reset().parameter(principalAmountParamName).value(principalAmount).notNull();
-        baseDataValidator.reset().parameter(principalAmountParamName).value(principalAmount).longGreaterThanZero();
+        BigDecimal principalAmount = this.fromApiJsonHelper.extractBigDecimalWithLocaleNamed(principalAmountParamName, element);
+        baseDataValidator.reset().parameter(principalAmountParamName).value(principalAmount).notNull()
+                .notLessThanMin(BigDecimal.valueOf(0));
```

Comment:
This can be changed to GreaterThanZero or isPositiveAmount. Accepting 0 principal seems not valid.